controllers/csi/provisioner/reconciler.go

```diff
@@ -149,8 +155,16 @@ func (r *OneAgentProvisioner) updateAgent(dk *dynatracev1alpha1.DynaKube, dtc dt
 
 	if ver != currentVersion {
 		if err := r.installAgentVersion(ver, envDir, dtc, logger); err != nil {
+			r.recorder.Eventf(dk,
+				corev1.EventTypeWarning,
+				failedInstallAgentVersionEvent,
+				"Failed to installed agent version: %s to envDir: %s, err: %s", ver, envDir, err)
 			return err
 		}
+		r.recorder.Eventf(dk,
+			corev1.EventTypeNormal,
+			installAgentVersionEvent,
+			"Installed agent version: %s to envDir: %s", ver, envDir)
```

Comment:
Message lacks the information about the node. This field must be available to be useful.